src/ui/component/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,49 @@
+import type { FC } from 'react'
+import { useCallback } from 'react'
+import classNames from 'classnames'
+import * as RCheckbox from '@radix-ui/react-checkbox'
+import { Icon } from '@/ui/component/icon/icon'
+import './checkbox.scss'
+
+export type CheckboxOption = {
+  id: string
+  name: string
+  checked: boolean
+  disabled: boolean
+}
+
+type CheckboxProps = {
+  id: string
+  name: string
```

Comment:
is the name important here, as we already have an `id`?

Reply:
Yes, checkbox `name` purpose is to provide a context about the checkbox value or id. [More details here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox).

Reply:
If it's a context thing, I would have marked name as optional